src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -10735,15 +10735,24 @@ int RGWRados::cls_bucket_head_async(const DoutPrefixProvider *dpp, const RGWBuck
 
 // uses information that the store has easy access to transition to the shard calculatoin logic
 void RGWRados::calculate_preferred_shards(const DoutPrefixProvider* dpp,
+					  const RGWBucketInfo& bucket_info,
 					  const uint64_t num_objs,
 					  const uint32_t num_source_shards,
 					  bool& need_resharding,
 					  uint32_t* suggested_num_shards)
 {
   const uint32_t max_dynamic_shards =
     uint32_t(cct->_conf.get_val<uint64_t>("rgw_max_dynamic_shards"));
-  const uint64_t max_objs_per_shard =
+  uint64_t max_objs_per_shard =
     cct->_conf.get_val<uint64_t>("rgw_max_objs_per_shard");
+
+  if (bucket_info.versioning_enabled()) {
```

Comment:
do we want this for versioning-suspended buckets too?

Reply:
Doesn't the `versioning_enabled()` ensure that the versioning is not suspended?
```cpp
bool versioning_enabled() const { return (versioning_status() & (BUCKET_VERSIONED | BUCKET_VERSIONS_SUSPENDED)) == BUCKET_VERSIONED; }
```

Reply:
my point is that versioning-suspended buckets have extra index entries too, so reshard should probably treat them the same as versioning-enabled. changing this to `versioned()` would cover both